crates/context/config/src/client/env/config/types/starknet.rs

```diff
@@ -0,0 +1,484 @@
+            blob: ApplicationBlob {
+                high: Felt::from_bytes_be_slice(blob_high),
+                low: Felt::from_bytes_be_slice(blob_low),
+            },
+            size: value.size,
+            source: value.source.into(),
+            metadata: value.metadata.into(),
+        }
+    }
+}
+
+impl From<crate::types::ApplicationId> for ApplicationId {
+    fn from(value: crate::types::ApplicationId) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationId {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+impl From<Repr<crate::types::ApplicationId>> for ApplicationId {
+    fn from(value: Repr<crate::types::ApplicationId>) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationId {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+impl From<&Repr<crate::types::ApplicationId>> for ApplicationId {
+    fn from(value: &Repr<crate::types::ApplicationId>) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationId {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+// Similar implementations for ApplicationBlob
+impl From<crate::types::BlobId> for ApplicationBlob {
+    fn from(value: crate::types::BlobId) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationBlob {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+impl From<Repr<crate::types::BlobId>> for ApplicationBlob {
+    fn from(value: Repr<crate::types::BlobId>) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationBlob {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+impl From<&Repr<crate::types::BlobId>> for ApplicationBlob {
+    fn from(value: &Repr<crate::types::BlobId>) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationBlob {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct EncodableString(pub String);
+
+impl From<crate::types::ApplicationSource<'_>> for EncodableString {
+    fn from(value: crate::types::ApplicationSource<'_>) -> Self {
+        EncodableString(value.0.into_owned())
+    }
+}
```

Comment:
this shouldn't need to be a `String`, we can't guarantee it's utf-8, it's pretty much just arbitrary bytes.

Reply:
hmm, I'm a little confused here to be honest, I thought the idea was for metadata and source to be "long Strings" in terms of types for Starknet since that's the only type that can fit, that is why this was implemented like this.

Reply:
so.. application source is a string, metadata, isn't. but on the starknet side, it's pretty much the same, dynamic array of felts.

but on the rust side, metadata is `Vec<u8>`